sdk/typescript/src/multisig/publickey.ts

```diff
@@ -0,0 +1,289 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { fromB64, toB64 } from '@mysten/bcs';
+import { blake2b } from '@noble/hashes/blake2b';
+import { bytesToHex } from '@noble/hashes/utils';
+import { PublicKey, bytesEqual } from '../cryptography/publickey.js';
+import type {
+	SerializedSignature,
+	SignatureFlag,
+	SignatureScheme,
+} from '../cryptography/signature.js';
+import {
+	SIGNATURE_FLAG_TO_SCHEME,
+	SIGNATURE_SCHEME_TO_FLAG,
+	parseSerializedSignature,
+} from '../cryptography/signature.js';
+import { normalizeSuiAddress } from '../utils/sui-types.js';
+import { builder } from '../builder/bcs.js';
+// eslint-disable-next-line import/no-cycle
+import { publicKeyFromBytes } from '../verify/index.js';
+import { bcs } from '../types/sui-bcs.js';
+
+type CompressedSignature =
+	| { ED25519: number[] }
+	| { Secp256k1: number[] }
+	| { Secp256r1: number[] };
+
+type PublicKeyEnum = { ED25519: number[] } | { Secp256k1: number[] } | { Secp256r1: number[] };
```

Comment:
Can we just share a type between `CompressedSignature` and this?

Reply:
After looking at this more, I would rather leave them as separate types.  This make anything more complex, and coming up with a name that makes sense for both Signatures and Public keys (TaggedBytes?) makes things more confusing.  I think trying to combine these feels like optimizing for the wrong thing